src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -1452,7 +1452,10 @@ private static IEnumerable<TResult> RepeatIterator<TResult>(TResult element, int
 
         public static IEnumerable<TResult> Empty<TResult>()
         {
-            return EmptyEnumerable<TResult>.Instance;
+            // This is a more memory-intensive empty enumerable that allocates a new enumerator each time.
```

Comment:
I don't believe this comment is accurate any more.  Enumerating an empty array should not allocate a new enumerator: https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Array.cs#L2717